### PR TITLE
Add Structured Logging ADR

### PR DESCRIPTION
**Pull request by nathaniel-may**
### Description

ADR for structured logging architecture

### Checklist

- [ ] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ ] I have run this code in development and it appears to resolve the stated issue
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change

### CODE REVIEW

File: docs/arch/adr-002-structured-logging-arch.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+#### How to output the data
+We decided to use the std lib logger because it was far more difficult than we expected to get to work properly. Documentation was lacking, and reading the source code wasn't a quick way to learn. The std lib logger was used mostly out of a necessity, and because many of the pleasantries you get from using a log library we had already chosen to do explicitly with functions in our nominal typing layer. Swapping out the std lib logger in the future should be an easy task should we choose to do it.
+
+## Status
+Completed
+
+## Consequences 
+Adding a new log event is more cumbersome than it was previously. Instead of writing the message at the log callsite, you must create a new concrete class in the event types. This is more opaque for new contributors. All user-facing log messages now live in one file which makes the job of conforming them much simpler. Because they are all nominally typed separately, it opens up the possibility to have log documentation generated from the type hints as well as outputting our logs in multiple human languages if we want to translate our messages.
```
Comment (nathaniel-may) [inline docs/arch/adr-002-structured-logging-arch.md:32]: I feel like the way I wrote this, we might want an "opportunities" section.

Comment (emmyoop) [reply docs/arch/adr-002-structured-logging-arch.md:32]: Consequences are both negative and positive.  Aren't opportunities a subset of consequences?  Per the [document linked in the README](https://cognitect.com/blog/2011/11/15/documenting-architecture-decisions)

```Consequences This section describes the resulting context, after applying the decision. All consequences should be listed here, not just the "positive" ones. A particular decision may have positive, negative, and neutral consequences, but all of them affect the team and project in the future.```

Comment (nathaniel-may) [reply docs/arch/adr-002-structured-logging-arch.md:32]: ahh good quote. I think that answers my question.

**Review by iknox-fa — APPROVED** [review]
Sorry I missed this review request last month @nathaniel-may .  This looks great, thanks for getting it out there.

**Review by leahwicz — APPROVED** [review]

**Pull request merged**